StoreKitUnitTests/TrialOrIntroPriceEligibilityCheckerTests.swift

```diff
@@ -0,0 +1,189 @@
+        mockBackend = MockBackend()
+        let mockOperationDispatcher = MockOperationDispatcher()
+        let identityManager = MockIdentityManager(mockAppUserID: "app_user")
+        let mockProductsManager = MockProductsManager()
+        trialOrIntroPriceEligibilityChecker =
+        TrialOrIntroPriceEligibilityChecker(receiptFetcher: receiptFetcher,
+                                            introEligibilityCalculator: mockIntroEligibilityCalculator,
+                                            backend: mockBackend,
+                                            identityManager: identityManager,
+                                            operationDispatcher: mockOperationDispatcher,
+                                            productsManager: mockProductsManager)
+    }
+
+    func testSK1CheckTrialOrIntroPriceEligibility() {
+        setupSK1()
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { _ in
+        }
+    }
+
+    func testSK1CheckTrialOrIntroPriceEligibilityFetchesAReceipt() {
+        setupSK1()
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { _ in
+        }
+
+        expect(self.receiptFetcher.receiptDataCalled).to(beTrue())
+    }
+
+    func testSK1EligibilityIsCalculatedFromReceiptData() throws {
+        setupSK1()
+        let stubbedEligibility = ["product_id": IntroEligibilityStatus.eligible]
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = (stubbedEligibility, nil)
+
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
+        let receivedEligibilities = try XCTUnwrap(maybeEligibilities)
+        expect(receivedEligibilities.count) == 1
+    }
+
+    func testSK1EligibilityIsFetchedFromBackendIfErrorCalculatingEligibility() throws {
+        setupSK1()
+        let stubbedError = NSError(domain: RCPurchasesErrorCodeDomain,
+                                   code: ErrorCode.invalidAppUserIdError.rawValue,
+                                   userInfo: [:])
+        mockIntroEligibilityCalculator.stubbedCheckTrialOrIntroductoryPriceEligibilityResult = ([:], stubbedError)
+
+        let stubbedEligibility = ["product_id": IntroEligibility(eligibilityStatus: IntroEligibilityStatus.eligible)]
+        mockBackend.stubbedGetIntroEligibilityCompletionResult = (stubbedEligibility, nil)
+        var completionCalled = false
+        var maybeEligibilities: [String: IntroEligibility]?
+        trialOrIntroPriceEligibilityChecker!.sk1CheckEligibility([]) { (eligibilities) in
+            completionCalled = true
+            maybeEligibilities = eligibilities
+        }
+
+        expect(completionCalled).toEventually(beTrue())
```

Comment:
this test is somewhat indirectly testing that the backend is called, since it forces an error, and then stubs the backend result and checks that the result is returned, correct? 
should we also add a more explicit check, like a call count, for the backend method?

Reply:
I see, yes, we should, because we are not guaranteeing that the completion was called due to another way that wasn't calling the backend. Good catch!